src/services/mongoose/schema.ts

```diff
@@ -296,6 +296,96 @@ AccountSchema.index({
 
 export const Account = mongoose.model<AccountRecord>("Account", AccountSchema)
 
+const QuizSchema = new Schema<QuizCompletedRecord>({
+  accountId: {
+    type: String,
+    ref: "Account",
+    required: true,
+  },
+  quizId: {
+    type: String,
+    required: true,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+})
+
+QuizSchema.index({ accountId: 1, quizId: 1 }, { unique: true })
+
+export const Quiz = mongoose.model<QuizCompletedRecord>("Quiz", QuizSchema)
+
+const SupportChatSchema = new Schema<SupportChatRecord>({
+  accountId: {
+    type: String,
+    ref: "Account",
+    required: true,
+  },
+  supportChatId: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+})
+
+export const SupportChat = mongoose.model<SupportChatRecord>(
+  "SupportChat",
+  SupportChatSchema,
+)
+
+const pointSchema = new Schema<LocationRecord>({
+  type: {
+    type: String,
+    enum: ["Point"],
+    required: true,
+  },
+  coordinates: {
+    type: [Number],
+    required: true,
+  },
+})
+
+const MerchantSchema = new Schema<MerchantRecord>({
+  id: {
+    type: String,
+    index: true,
+    unique: true,
+    sparse: true,
+    required: true,
+    default: () => crypto.randomUUID(),
```

Comment:
is there a reason we're creating an id field rather than using Mongo's native `_id: ObjectID`?

Reply:
legacy code from Galoy